package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua

```diff
@@ -52,4 +53,29 @@ function M.is_outdoor_device()
 	return false
 end
 
+function M.supports_wpa3()
+	return unistd.access('/lib/gluon/features/wpa3')
+end
+
+function M.supports_mfp(uci)
+	local idx = 0
+	local supports_mfp = true
+
+	if not M.supports_wpa3() then
+		return false
+	end
+
+	uci:foreach('wireless', 'wifi-device', function()
+		local phypath = '/sys/kernel/debug/ieee80211/phy' .. idx .. '/'
+
+		if not util.file_contains_line(phypath .. 'hwflags', 'MFP_CAPABLE') then
```

Comment:
Ugh, it sucks that there is no robust way to determine this without debugfs ...

Reply:
Besides checking the debug we can screenscrape `iw` output.

```
iw phy phy0 info | grep 00-0f-ac:6
```

via https://kevinlocke.name/bits/2019/12/28/checking-802.11w-support/
https://kevinlocke.name/bits/2019/12/28/checking-802.11w-support/

Reply:
I know the debugfs is not the must robust way, but i doubt screenscraping iw (which itself says you should not screenscrape it) is the better solution.